ecs/pod_creator.go

```diff
@@ -492,150 +489,146 @@ func translateContainerStatusInfo(containers []*ecs.Container) []cocoa.ECSContai
 func exportPodDefinitionOptions(opts cocoa.ECSPodDefinitionOptions) *ecs.RegisterTaskDefinitionInput {
 	var taskDef ecs.RegisterTaskDefinitionInput
 
-	taskDef.SetContainerDefinitions(exportContainerDefinitions(opts.ContainerDefinitions)).
-		SetFamily(utility.FromStringPtr(opts.Name)).
-		SetTags(ExportTags(opts.Tags))
+	taskDef.ContainerDefinitions = exportContainerDefinitions(opts.ContainerDefinitions)
+	taskDef.Family = opts.Name
+	taskDef.Tags = ExportTags(opts.Tags)
+	taskDef.TaskRoleArn = opts.TaskRole
+	taskDef.ExecutionRoleArn = opts.ExecutionRole
 
 	if mem := utility.FromIntPtr(opts.MemoryMB); mem != 0 {
-		taskDef.SetMemory(strconv.Itoa(mem))
+		taskDef.Memory = aws.String(strconv.Itoa(mem))
 	}
 
 	if cpu := utility.FromIntPtr(opts.CPU); cpu != 0 {
-		taskDef.SetCpu(strconv.Itoa(cpu))
-	}
-
-	if opts.TaskRole != nil {
-		taskDef.SetTaskRoleArn(*opts.TaskRole)
-	}
-	if opts.ExecutionRole != nil {
-		taskDef.SetExecutionRoleArn(*opts.ExecutionRole)
+		taskDef.Cpu = aws.String(strconv.Itoa(cpu))
 	}
 
 	if opts.NetworkMode != nil {
-		taskDef.SetNetworkMode(string(*opts.NetworkMode))
+		taskDef.NetworkMode = types.NetworkMode(*opts.NetworkMode)
 	}
 
 	return &taskDef
 }
 
 // exportContainerDefinition converts container definitions into their
 // equivalent ECS container definition.
-func exportContainerDefinitions(defs []cocoa.ECSContainerDefinition) []*ecs.ContainerDefinition {
-	var containerDefs []*ecs.ContainerDefinition
+func exportContainerDefinitions(defs []cocoa.ECSContainerDefinition) []types.ContainerDefinition {
+	var containerDefs []types.ContainerDefinition
 
 	for _, def := range defs {
-		var containerDef ecs.ContainerDefinition
+		var containerDef types.ContainerDefinition
```

Comment:
Same as above - just use regular struct initialization and set the optional fields (memory, CPU, working dir) after the struct is initialized.

Reply:
done